src/Sylius/Bundle/CoreBundle/Routing/Generator/ChannelProductUrlGenerator.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Routing\Generator;
+
+use Doctrine\Common\Collections\Collection;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Core\Model\ProductTranslationInterface;
+use Sylius\Component\Locale\Context\LocaleContextInterface;
+use Sylius\Component\Locale\Model\LocaleInterface;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+final class ChannelProductUrlGenerator implements ChannelProductUrlGeneratorInterface
```

Comment:
```suggestion
final class ShopProductUrlChannelBasedGenerator implements ShopProductUrlChannelBasedGeneratorInterface
```
Or sth similar, because I am not convinced about this one, but I would put the keyword "shop" somewhere to indicate precisely what kind of URL it  generates